src/components/PostDetails.tsx

```diff
@@ -1,107 +1,126 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment, getComments } from '../api/api';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+export interface Props {
+  post: Post;
+}
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [showForm, setShowForm] = useState(false);
 
-        <div className="block">
-          <Loader />
+  const title = `#${post.id}: ${post.title}`;
+  const showButton = !isLoading && !showForm && !isError;
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getComments(post)
+      .then(setComments)
+      .catch(() => setIsError(true))
```

Comment:
The error handling here is minimal. Consider providing more detailed error messages or logging to help with debugging and user feedback.